src/backend/distributed/transaction/distributed_deadlock_detection.c

```diff
@@ -175,32 +175,46 @@ CheckForDistributedDeadlocks(void)
 			{
 				TransactionNode *currentNode =
 					(TransactionNode *) lfirst(participantTransactionCell);
+				bool transactionAssociatedWithProc =
+					AssociateDistributedTransactionWithBackendProc(currentNode);
+				TimestampTz youngestTimestamp = 0;
+				TimestampTz currentTimestamp = 0;
 
-				TimestampTz youngestTimestamp =
-					youngestTransaction->transactionId.timestamp;
-				TimestampTz currentTimestamp = currentNode->transactionId.timestamp;
+				LogTransactionNode(currentNode);
 
-				AssociateDistributedTransactionWithBackendProc(currentNode);
+				/* we couldn't find the backend process originated the transaction */
+				if (!transactionAssociatedWithProc)
+				{
+					continue;
+				}
 
-				LogTransactionNode(currentNode);
+				if (youngestTransaction == NULL)
+				{
+					youngestTransaction = currentNode;
+					continue;
+				}
 
-				if (currentNode->transactionId.initiatorNodeIdentifier ==
-					GetLocalGroupId() &&
-					timestamptz_cmp_internal(currentTimestamp, youngestTimestamp) == 1)
+				youngestTimestamp = youngestTransaction->transactionId.timestamp;
+				currentTimestamp = currentNode->transactionId.timestamp;
+				if (timestamptz_cmp_internal(currentTimestamp, youngestTimestamp) == 1)
 				{
 					youngestTransaction = currentNode;
 				}
 			}
 
-			/* we should find the backend */
-			Assert(youngestTransaction->initiatorProc != NULL);
-
-			CancelTransactionDueToDeadlock(youngestTransaction->initiatorProc);
-			LogCancellingBackend(youngestTransaction);
+			/*
+			 * We found the deadlock and its associated proc is still
+			 * alive.
+			 */
+			if (youngestTransaction && youngestTransaction->initiatorProc != NULL)
```

Comment:
Hmm, that can never be `NULL`, but it could have `pid < 0`, which should be fine since we already have a `warning` (i.e., we can remove the second part of the `if` statement?)